api/helpers.go

```diff
@@ -91,3 +92,18 @@ func (a *API) requestAud(ctx context.Context, r *http.Request) string {
 	// Finally, return the default of none of the above methods are successful
 	return config.JWT.Aud
 }
+
+func (a *API) getReferrer(r *http.Request) string {
+	ctx := r.Context()
+	config := a.getConfig(ctx)
+	referrer := ""
+	if reqref := r.Referer(); reqref != "" {
+		base, berr := url.Parse(config.SiteURL)
+		refurl, rerr := url.Parse(reqref)
+		// As long as the referrer came from the site, we will redirect back there
```

Comment:
What about localhost?

Reply:
What do you mean?

Reply:
If you try to log in from a site running on localhost, will the confirm emails pick that up in the redirect they send you?

Reply:
Only if the site URL is also localhost